### PR TITLE
tests: Refactor testing framework

### PR DESCRIPTION
**Pull request by LecrisUT**
This PR simply moves the tests around to either `unit_tests`, `integration_tests` (currently none) or `functional_tests`.

TODO:
- [x] Check if there are any other non-functional tests in the list

### CODE REVIEW

File: test/functional/c/CMakeLists.txt [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+		error.cpp
+		symmetry.cpp
+		symmetry_search.cpp
+		spacegroup_type_search.cpp
+		find_primitive_cell.cpp
+		refine_cell.cpp
+		standardization.cpp
+		dataset_access.cpp
+		magnetic_symmetry.cpp
+		kpoints.cpp
```
Comment (lan496) [inline test/functional/c/CMakeLists.txt]: Why do you rename `test_kpoints.cpp` to `kpoints.cpp` and so on? Is there any correspoding style guide? If not, I prefer not to change these file names.

**Review by lan496 — CHANGES_REQUESTED** [review]
Can you explain the difference between `test/functional` and `test/unit`?

**Comment by LecrisUT** [comment]
> Can you explain the difference between `test/functional` and `test/unit`?

I wrote it in another PR, but:
<details><summary>Unit/integration/functional test</summary>

 ### unit test
- runs a single self-contained internal function, e.g. constructing and destructing the lattice object
- it does not touch any physical aspects
- used to debug compilation, segmentation issues etc.
- the main thing to investigate in codecov

### functional test
- runs the full calculation as the user would be doing
- requires input data
- tests the physical aspects of the library

### integration test (yes, there's one more technically)
- middle step between unit and functional test. Runs multiple fumctions covered by unit tests or the higher level functions
- tests if the data is transfered correctly between the functions, mainly for segmentation faults and unexpected crashes in the manipulation of data

</details>

So in this case, because most of the tests are calling through `spglib.h` directly most of them are functional tests (what the user would be using)
- Example of unit-tests: constructor/destructor, individual maths operations, individual operations like search symmetry (constructors/destructors must be handled by the test fixture, not the api)
- Example of functional test: calling api functions like `spg_get_hall_number_from_symmetry`

There are edge-cases like `spg_get_spacegroup_type` where it is an api, but it only calls constructors, and because there is no complicated logic that calls other units, I consider this as a unit test.

File: test/functional/c/CMakeLists.txt (hunk 1 above)
Comment (LecrisUT) [reply test/functional/c/CMakeLists.txt]: Not really, it's just more consistent with the naming of the test-suite. But I do get the appeal that it separates the tests from the unity libraries

**Comment by lan496** [comment]
Thanks for your explanation. I confused the distinction between `unit` and `functional`.

**Review by lan496 — APPROVED** [review]

**Comment by lan496** [comment]
@atztogo LGTM. Do you have any comment? This PR touches a lot of files, so I think it is necessary to consult on you.

**Review by atztogo — APPROVED** [review]
Thanks @LecrisUT and @lan496. I confirmed pytest works on my laptop, by which I am satisfied due to your agreement on other tests. Reviewing more than that requires me to take time to understand, but no problem for me to let it go than blocking.

**Comment by LecrisUT** [comment]
Yeah, but this PR only moves and renames things so it's not effecting anything atm. #303 is where things start changing
